okhttp-tests/src/test/java/okhttp3/internal/http2/HttpOverH2CTest.java

```diff
@@ -0,0 +1,85 @@
+package okhttp3.internal.http2;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Protocol;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static okhttp3.TestUtil.defaultClient;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test H2C, the cleartext implementation of HTTP/2.
+ *
+ * @author jpitz
+ */
+@SuppressWarnings("Duplicates")
+public final class HttpOverH2CTest extends Http2TestBase {
+  private static final List<Protocol> PROTOCOLS = Arrays.asList(Protocol.H2C);
+
+  @Before
+  public void setUp() throws Exception {
+    super.setUp(
+      defaultClient().newBuilder()
+          .protocols(PROTOCOLS)
+          .build(),
+      "http",
+      Protocol.H2C
+    );
+    server.setProtocols(PROTOCOLS);
+  }
+
+  @Test
+  public void testProtocol() throws Exception {
```

Comment:
It feels like these tests from here onwards would be more natural in ProtocolTest (new), MockWebserverTest and OkHttpClientTest.  Thoughts?

Reply:
Ah. I looked for a ProtocolTest and there wasn't one. Didn't look again after handling the merge. I'll move them.